test.js

```diff
@@ -50,6 +50,30 @@ test('register', function (t) {
     })
   })
 
+  t.test('secret as a promise', function (t) {
+    t.plan(1)
+    const fastify = Fastify()
+    fastify.register(jwt, {
+      secret: async function () {
+        return await Promise.resolve('supersecret')
+      }
```

Comment:
this is a by-product of how you implemented it. the function needn't be async (but you should also support async), it should be able to return a promise and it should be handled even when the function is not async. also note that awaiting the return value of an async function is an anti pattern. see https://www.nearform.com/blog/javascript-promises-the-definitive-guide/

Reply:
Dropped async